app/models/rbac.rb

```diff
@@ -124,6 +124,16 @@ def self.get_self_service_object_ids(user_or_group, klass)
     targets
   end
 
+  def self.calc_filtered_ids(klass, scope, user_filters, user_or_group = nil)
```

Comment:
Is user_filters a good name here?  I'm concerned if we pass other filters, this code becomes confusing...